packages/commerce-sdk-react/src/hooks/useMutation.ts

```diff
@@ -76,40 +77,42 @@ export const useCustomMutation = <TData = unknown, TError = unknown>(
 ) => {
     const auth = useAuthContext()
     const globalConfig = useConfig()
-    const globalHeaders = globalConfig.headers || {}
-    const globalClientConfig = {
-        parameters: {
-            clientId: globalConfig.clientId,
-            siteId: globalConfig.siteId,
-            organizationId: globalConfig.organizationId,
-            shortCode: globalConfig.shortCode
-        },
-        proxy: globalConfig.proxy
-    }
+    const logger = globalConfig.logger || console
 
     const createMutationFnWithAuth = (): MutationFunction<TData, TMutationVariables> => {
         return async (args): Promise<TData> => {
             const {access_token} = await auth.ready()
-            return (await helpers.callCustomEndpoint({
-                ...apiOptions,
-                options: {
-                    ...apiOptions.options,
-                    headers: {
-                        Authorization: `Bearer ${access_token}`,
-                        // Note the order of the following destructred objects is important.
-                        // Priority assending order: global config < mutation config < mutate func args
-                        ...globalHeaders,
-                        ...apiOptions.options?.headers,
-                        ...(args?.headers ? args.headers : {})
-                    },
-                    ...(args?.body ? {body: args.body} : {}),
-                    ...(args?.parameters ? {parameters: args.parameters} : {})
-                },
-                clientConfig: {
-                    ...globalClientConfig,
-                    ...(apiOptions.clientConfig || {})
-                }
-            })) as TData
+            return (await helpers
+                .callCustomEndpoint(
+                    generateCustomEndpointOptions(args, apiOptions, globalConfig, access_token)
+                )
+                .catch(async (error) => {
+                    if (error?.response?.status == 401) {
+                        const response = await error?.response?.json()
+                        if (
+                            response?.detail ===
+                            'Customer credentials changed after token was issued.'
+                        ) {
+                            logger.info('Login was invalidated. Clearing login state.')
+                            await auth.logout()
+
+                            // Retry again after resetting auth state
+                            const {access_token} = await auth.ready()
```

Comment:
this implementation is different from `useAuthorizationHeader` where it does

```js
const {access_token} = await auth.logout()
```

and here it has 2 awaits, this seems inconsistent

Reply:
it would be nice if the code is more DRY

Reply:
I've made changes to clean up and reduce repetition. Could you please review again @kevinxh